### ISSUE TITLE
DiskSpaceHealthIndicator  : Free disk space below threshold. Available: 0 bytes, acctually disk space is 80GB.

### ISSUE THREAD

**Issue by dodolzg** [spring-projects/spring-boot#11058]
The version of spring-boot is 1.3.8.RELEASE

Here is warning log :

> WARN 24238 --- [0.1-9000-exec-6] o.s.b.a.health.DiskSpaceHealthIndicator  : Free disk space below threshold. Available: 0 bytes (threshold: 10485760 bytes)

Actually,  the disk info :

> Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1        99G   14G   80G  15% /
devtmpfs        3.9G     0  3.9G   0% /dev
tmpfs           3.9G     0  3.9G   0% /dev/shm
tmpfs           3.9G  536K  3.9G   1% /run
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
tmpfs           783M     0  783M   0% /run/user/0
tmpfs           783M     0  783M   0% /run/user/1000

> Disk /dev/vda: 107.4 GB, 107374182400 bytes, 209715200 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk label type: dos
Disk identifier: 0x0008e3b4
   Device Boot      Start         End      Blocks   Id  System
/dev/vda1   *        2048   209713151   104855552   83  Linux

When I reboot the application, it works fine. Why does DiskSpaceHealthIndicator sometimes  show warning log?

**Comment by snicoll** [comment]
@dodolzg it is hard to say but I suspect something in your report is missing. We're looking at the available disk space for the working directory of the app. Double check that location and your environment (the reboot part is suspicious).

`management.health.diskspace.path` allows you to give an alternate path. Can you please try that?

**Comment by dodolzg** [comment]
The location of app is "/data/webapp", which is in "/dev/vda1", and the space of this disk is 80G

**Comment by wilkinsona** [comment]
The health indicator calls `File.getUsableSpace()`. The return value of that method is the `0` in this warning message:

```
Free disk space below threshold. Available: 0 bytes (threshold: 10485760 bytes)
```

In other words, for that warning message to appear, `File.getUsableSpace()` must have returned 0. This is out of Spring Boot's control and, largely, out of the JVM's control too as it just makes a call to the underlying filesystem. 

One possible cause is that something filled `/dev/vda1` that was cleaned up when you rebooted. Depending on your Linux distro and how's it's configured that may be how `/tmp` behaves.

I'm going to close this one for now as I can't see any evidence that it's a bug in Spring Boot. If you find such evidence then please let us know and we'll re-open the issue.

**Comment by wilkinsona** [comment]
A minor correction to what I wrote above: on 1.3.8 we can `File.getFreeSpace()` rather than `File.getUsableSpace()`.

**Comment by whh19921024** [comment]
I met this problem,Finally found is the current directory Permission denied, Then switch to the other directory,It is ok to start.

**Comment by HikoQiu** [comment]
I met the same problem too.

The warning logs:

`2018-04-25 01:53:09,884 WARN org.springframework.boot.actuate.health.DiskSpaceHealthIndicator [DiscoveryClient-InstanceInfoReplicator-0] Free disk space below threshold. Available: 0 bytes (threshold: 10485760 bytes)`
`2018-04-25 01:53:09,886 WARN com.netflix.discovery.DiscoveryClient$3 [DiscoveryClient-InstanceInfoReplicator-0] Saw local status change event StatusChangeEvent [timestamp=1524592389886, current=DOWN, previous=UP] `


Disk space of the host:
`Filesystem  Size Used Avail Use% Mounted on`
`/dev/vda1    40G  12G  27G   31%   /`
